### PR TITLE
Include tests for module 1 exercises. Possible typos

### PR DESCRIPTION
**Pull request by jkim222383**
@hfboyce I don't think I have permission to push to this repository yet. I have temporarily forked this repository.
 
@ecool50 Could you review the tests and suggest any changes needed?

Some notes:

- For Exercise 01_06, there's a minor typo in the instruction on the exercise page: 
> ... Altair plotin order to ...

should be changed to

> ... Altair plot in order to ...

- `solution_01_06.py` has `Horsepower` on y-axis even though instructions on the exercise page state `Horsepower` on the x-axis.

I chose to go with the solution version when writing `test_01_06.py` because I’m assuming the typo is in the instructions, not the solution. I think the instructions were meant to state

> The cars’ Weight_in_lbs should be on the x-axis and their Horsepower on the y-axis.

instead of

> The cars’ Horsepower should be on the x-axis and their Weight_in_lbs on the y-axis.

but if the instructions are correct and the solution file is the wrong version, I can switch the axes in both the solution file and the test file.

### CODE REVIEW

File: exercises/en/test_01_06.py [hunk 1]
```diff
@@ -8,6 +8,8 @@ def test():
     # Since we haven't started assigning charts to variable names yet,
     # this might be the better way to test for the first exercise.
     # Maybe even for later exercises.
+    assert 'alt.Chart(cars)' in __solution__, "Make sure you're passing the correct data object for plotting."
+    assert 'mark_point(' in __solution__, "Make sure you're using the correct type of mark"
```
Comment (ecool50) [inline exercises/en/test_01_06.py:12]: I personally don't think you need the `(` here, but it should be fine.

File: exercises/en/test_01_09a.py [hunk 2]
```diff
@@ -8,6 +8,11 @@ def test():
     # Since we haven't started assigning charts to variable names yet,
     # this might be the better way to test for the first exercise.
     # Maybe even for later exercises.
+    assert 'from vega_datasets import data' in __solution__, "Did you import the correct library for data from vega_datasets?"
```
Comment (ecool50) [inline exercises/en/test_01_09a.py:11]: Be careful here. Sometimes the students are not careful with white spaces.

File: exercises/en/test_01_09a.py [hunk 3]
```diff
@@ -8,6 +8,11 @@ def test():
...
+    assert 'data.stocks()' in __solution__, "Make sure to call the stocks data from the data library"
+    assert '# alt.Chart' not in __solution__, "Make sure you remove the hash (#) symbol used to comment out the plotting codes."
```
Comment (ecool50) [inline exercises/en/test_01_09a.py:13]: Nice!

File: exercises/en/test_01_09b.py [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
 
-    # Since we haven't started assigning charts to variable names yet,
-    # this might be the better way to test for the first exercise.
-    # Maybe even for later exercises.
-
-    # TODO implement test
-    __msg__.good("Not yet implemented")
-    # __msg__.good("You're correct, well done!")
+    # Regex expression to capture the line plot variable name without whitespace
+    capture_grp = re.search("[^\s]*(?=\s*=\s*alt.Chart\(stocks\).mark_line)", __solution__)
```
Comment (ecool50) [inline exercises/en/test_01_09b.py:11]: Nice!

File: exercises/en/test_01_09b.py [hunk 5]
```diff
@@ -1,14 +1,15 @@
...
+    assert capture_grp, "Make sure you assign the line plot to a variable name."
+    # capture_grp.group() holds the variable name of line plot 
+    assert capture_grp.group() + ".mark_point(" in __solution__, "Make sure you add a layer of another plot object, created from the line plot object with the same encodings but different mark type."
```
Comment (ecool50) [inline exercises/en/test_01_09b.py:14]: Nice!

File: exercises/en/test_01_10.py [hunk 6]
```diff
@@ -9,6 +9,13 @@ def test():
     # this might be the better way to test for the first exercise.
     # Maybe even for later exercises.
 
-    # TODO implement test
-    __msg__.good("Not yet implemented")
-    # __msg__.good("You're correct, well done!")
+    assert 'data.stocks()' in __solution__, ""
+    assert 'alt.Chart(stocks)' in __solution__, "Make sure you're passing the correct data object fot plotting."
+    assert 'mark_line(' in __solution__, "Make sure you're using the correct type of mark"
```
Comment (ecool50) [inline exercises/en/test_01_10.py:14]: `mark_line(` or `mark_line`? Be consistent.

File: exercises/en/test_01_10.py [hunk 7]
```diff
@@ -9,6 +9,13 @@ def test():
...
+    assert 'x="date"' in __solution__, "Did you encode the x channel correctly?"
+    # More specific hint
+    assert 'y="price"' not in __solution__, "Almost there, but not quite. Here, you want one line with the average value of all stocks over time. Can you modify the encoding for y to aggregate all prices into a single mean?"
+    assert 'y="mean(price)"' in __solution__, "Did you encode the y channel correctly?"
```
Comment (ecool50) [inline exercises/en/test_01_10.py:18]: Maybe ask them if they are averaging somehow?

**Review by ecool50 — COMMENTED** [review]
Overall I think this was well done. Very minor (almost pedantic comments). 
Your turn @hfboyce.

File: exercises/en/test_01_09a.py (hunk 2 above)
Comment (hfboyce) [reply exercises/en/test_01_09a.py:11]: I would instead just check that "data" or "import data"  is in `__solution__`.  Agreed that white space is frustrating for students and checking less, is more robust sometimes.

File: exercises/en/test_01_09a.py (hunk 3 above)
Comment (hfboyce) [reply exercises/en/test_01_09a.py:13]: I love the # alt.Chart, that's something Elijah and I have not done before!

File: exercises/en/test_01_09a.py [hunk 8]
```diff
@@ -8,6 +8,11 @@ def test():
...
+    assert 'alt.Chart(stocks)' in __solution__, "Make sure you're passing the correct data object fot plotting."
+    assert 'mark_line' in __solution__, "Make sure you're using the correct type of mark"
     assert 'x="date"' in __solution__, "Did you encode the x channel correctly?"
     assert 'y="price"' in __solution__, "Did you encode the y channel correctly?"
     assert 'color="symbol"' in __solution__, "Did you encode the color channel correctly?"
```
Comment (hfboyce) [inline exercises/en/test_01_09a.py:18]: I'm curious if we can make these more robus by checking x="date" OR `x= "date"` to account for the white space issue? (for x, y, and color)

File: exercises/en/test_01_09b.py (hunk 4 above)
Comment (hfboyce) [reply exercises/en/test_01_09b.py:11]: One small comment, and (@ecool50 can attest to this) is I usually like as mix of both testing if a string is in __solution__ as well as checking the object itself. The last few questions we did not assign to a variable but when we do I like a mix of the two. Just something to keep in mind.

File: exercises/en/test_01_09b.py (hunk 5 above)
Comment (hfboyce) [reply exercises/en/test_01_09b.py:14]: These are new for us. Cool!

**Review by hfboyce — CHANGES_REQUESTED** [review]

**Comment by hfboyce** [comment]
> For Exercise 01_06, there's a minor typo in the instruction on the exercise page

Great!, Also fixed! Nice catch! 

> solution_01_06.py has Horsepower on y-axis even though instructions on the exercise page state Horsepower on the x-axis.
I chose to go with the solution version when writing test_01_06.py because I’m assuming the typo is in the instructions, not the solution. I think the instructions were meant to state.

Yes! No changes that you need to do for that. I've already fixed the instructions. Thanks :)

File: exercises/en/test_01_09a.py (hunk 8 above)
Comment (jkim222383) [reply exercises/en/test_01_09a.py:18]: For this plot in `01_09a`, I couldn't figure out a way to do it, but I've modified the test for `01_09b` to check that the attributes for the plot object that was given in the sample code remain unmodified (as a practice for future exercises where students will bind names to the plots )

**Comment by jkim222383** [comment]
Thank you both for the feedback. Suggested changes were made in:
fbe4fdd89549995cbe8607f76cfd3a0a6cb01bb1 (most of the suggested changes)
9ea18d26de1dced83d30a165f0563600e2dbf15d (to make `test_01_09b.py` more robust in case the variable name matched using regex no longer holds a Chart object)

**Pull request merged**